src/coreclr/vm/riscv64/stubs.cpp

```diff
@@ -1525,68 +1555,452 @@ static void LoadRegPair(BYTE* p, int reg1, int reg2, UINT32 offset)
+            p += 4;
+        }
+
+        BYTE* pBLTCall = NULL;
+
+        for (WORD i = 0; i < pLookup->indirections; i++)
+        {
+            if (i == pLookup->indirections - 1 && pLookup->sizeOffset != CORINFO_NO_SIZE_CHECK)
+            {
+                _ASSERTE(pLookup->testForNull && i > 0);
+
+                if (pLookup->sizeOffset > 2047)
+                {
+                    *(DWORD*)p = ITypeInstr(0x3, 0x2, RegT4, RegT4, dataOffset);// lw  t4, dataOffset(t4)
+                    p += 4;
+                    *(DWORD*)p = RTypeInstr(0x33, 0, 0, RegT5, RegA0, RegT4);// add  t5, a0, t4
+                    p += 4;
+                    *(DWORD*)p = ITypeInstr(0x3, 0x3, RegT5, RegT5, 0);// ld  t5, 0(t5)
+                    p += 4;
+
+                    // move to next indirection offset data
+                    dataOffset += 4;
+                }
+                else
+                {
+                    *(DWORD*)p = ITypeInstr(0x3, 0x3, RegT5, RegA0, (UINT32)pLookup->sizeOffset);// ld  t5, #(pLookup->sizeOffset)(a0)
+                    p += 4;
+                }
+                // lui  t4, (slotOffset&0xfffff000)>>12
+                *(DWORD*)p = UTypeInstr(0x37, RegT4, (((UINT32)slotOffset & 0xfffff000) >> 12));
+                p += 4;
+                *(DWORD*)p = ITypeInstr(0x13, 0, RegT4, RegT4, slotOffset & 0xfff);// addi  t4, t4, (slotOffset&0xfff)
+                p += 4;
+                // blt  t4, t5, CALL HELPER
+                pBLTCall = p;       // Offset filled later
+                *(DWORD*)p = BTypeInstr(0x63, 0x4, RegT4, RegT5, 0);
+
+                p += 4;
+            }
+
+            if(pLookup->offsets[i] > 2047)
+            {
+                _ASSERTE(dataOffset < 2047);
+                *(DWORD*)p = ITypeInstr(0x3, 0x2, RegT4, RegT4, dataOffset & 0xfff);// lw  t4, dataOffset(t4)
+                p += 4;
+                *(DWORD*)p = RTypeInstr(0x33, 0, 0, RegA0, RegA0, RegT4);// add  a0, a0, t4
+                p += 4;
+                *(DWORD*)p = ITypeInstr(0x3, 0x2, RegA0, RegA0, 0);// lw  a0, 0(a0)
+                p += 4;
+                // move to next indirection offset data
+                dataOffset += 4; // add 4 as next data is at 4 bytes from previous data
+            }
+            else
+            {
+                // offset must be 8 byte aligned
+                _ASSERTE((pLookup->offsets[i] & 0x7) == 0);
+                *(DWORD*)p = ITypeInstr(0x3, 0x3, RegA0, RegA0, (UINT32)pLookup->offsets[i]);// ld  a0, #(pLookup->offsets[i])(a0)
+                p += 4;
+            }
+        }
+
+        // No null test required
+        if (!pLookup->testForNull)
+        {
+            _ASSERTE(pLookup->sizeOffset == CORINFO_NO_SIZE_CHECK);
+            *(DWORD*)p = ITypeInstr(0x67, 0, RegR0, RegRa, 0);// ret
+            p += 4;
+        }
+        else
+        {
+            // beq  a0, x0, CALL HELPER:
+            *(DWORD*)p = BTypeInstr(0x63, 0, RegA0, RegR0, 0);
```

Comment:
I think imm value should not be 0. I think it is `8`?? Please check again.

Reply:
Fixed it. Thank you!